File: packages/typescript/src/territorial/generateRecommendations.ts
import { recommendationFor } from "./recommendationFor";

export function generateRecommendations(entities: string[]): string[] {
  const seen = new Set<string>();
  const result: string[] = [];
  for (const entity of entities) {
    if (seen.has(entity)) {
      continue;
    }
    seen.add(entity);
    result.push(recommendationFor(entity));
  }
  return result;
}

File: packages/dart/lib/src/territorial/generate_recommendations.dart
import 'recommendation_for.dart';

List<String> generateRecommendations(List<String> entities) {
  final seen = <String>{};
  final result = <String>[];
  for (final entity in entities) {
    if (seen.add(entity)) {
      result.add(recommendationFor(entity));
    }
  }
  return result;
}
